contracts/Payroll.sol

```diff
@@ -223,29 +260,46 @@ contract Payroll is Initializable, OwnableUpgradeable, ReentrancyGuardUpgradeabl
             // return the leftover of _erc20TokenOrigin
             TransferHelper.safeTransfer(_erc20TokenOrigin, msg.sender, leftOver);
         }
-        return totalAmountIn;
     }
 
-    function toAddress(bytes memory _bytes, uint256 _start) internal pure returns (address) {
-        require(_start + 20 >= _start, "toAddress_overflow");
-        require(_bytes.length >= _start + 20, "toAddress_outOfBounds");
-        address tempAddress;
+    function _performSwapV3ETH(
+        uint256 _totalAmountToSwap,
+        uint32 _deadline,
+        SwapV3[] calldata _swaps
+    ) internal returns (uint256) {
+        require(msg.value >= _totalAmountToSwap, "Payroll: Not enough msg.value");
+        address weth = IUniswapV3(swapRouter).WETH9();
 
-        assembly {
-            tempAddress := div(mload(add(add(_bytes, 0x20), _start)), 0x1000000000000000000000000)
+        uint256 totalAmountIn = 0;
+        for (uint256 i = 0; i < _swaps.length; i++) {
+            require(_swaps[i].path.length > 0, "Payroll: Empty path");
+            uint256 amountIn = IUniswapV3(swapRouter).exactOutput{value: _swaps[i].amountInMax}(
```

Comment:
Add a comment pointing to https://soliditydeveloper.com/uniswap3 as the oficial documentation does not say anything about the value or the return ether